### DISCUSSION TITLE
Enhancing useRoute to use RouteRecordInfo meta type from RouteNamedMap

### DISCUSSION THREAD

**Discussion by joris-gallot** [posva/unplugin-vue-router#389] [Ideas]
I have a suggestion regarding the typing of `useRoute`, specifically concerning the `meta` property.

Currently, `useRoute` returns [RouteLocationNormalizedLoadedTyped](https://github.com/posva/unplugin-vue-router/blob/8b3e1f028c521d7a3f1e4b068411e4a0a0f38577/src/type-extensions/routeLocation.ts#L32), which does not leverage the `meta` type defined in the `RouteNamedMap`. I believe there could be a more type-safe way to handle the `meta` property by incorporating it into `RouteLocationNormalizedLoadedTyped`

Here is a proposed enhancement:

```typescript
interface RouteLocationNormalizedLoadedTyped<
  RouteMap extends _RouteMapGeneric = _RouteMapGeneric,
  Name extends keyof RouteMap = keyof RouteMap,
> extends RouteLocationNormalizedLoaded { {
    name: Extract<Name, RouteRecordName>;
    params: RouteMap[Name]['params'];
    meta: RouteMap[Name]['meta'];
}
```

By default, `RouteMap[Name]['meta']` would have the type [RouteMeta](https://github.com/posva/unplugin-vue-router/blob/8b3e1f028c521d7a3f1e4b068411e4a0a0f38577/src/codegen/generateRouteMap.ts#L46), preserving the current behavior while allowing for the `meta` type to be overridden in `RouteRecordInfo`

#### Example
```ts
interface RouteNamedMap {
    Home: RouteRecordInfo<'Home', '/', RouteParamsRaw, RouteParams, { foo: 'bar' }>;
	...
}

const route = useRoute<'Home'>()
route.meta // { foo: 'bar' }
```

**Comment by posva** [comment]
This is planned! Ideally we could even extract the type from definePage otherwise it could be passed as type parameter to a a generic definePage

**Reply by joris-gallot** [reply]
Thanks for the quick response! I'm generating the `RouteNamedMap` myself to use with the plugin, it's good to know that this feature is coming; I think I'll use a fork until these type changes are implemented, thanks!
